### ISSUE TITLE
Olex2

### ISSUE THREAD

**Issue by SamLukeYes** [archlinuxcn/repo#1965]
### 问题类型 / Type of issues

* 软件打包请求 / package request

----

* 简述 / Brief Description

    Olex² is a simple to use program containing everything you need to solve, refine and finish small-molecule crystal structures using an intuitive user interface, honed over ten years of development.

* 主要语言和工具链 / Main Languages and Toolchains

    gcc, python

* 上游链接 / Upstream Link

    Main Page: https://www.olexsys.org

    Source Repo: https://github.com/pcxod/olex2

    Binary Releases: http://www.olex2.org/olex2-distro/

* 为何需要这个软件包 / Why Need This Package

    It is a convenient tool to solve crystal structures, and is required in my class. According to my teacher, it is widely used in chemical research. However, downloading from the official website is extremely slow in mainland China, and I didn't find anything about it on Arch Linux Forum and AUR, while it does provide releases for Linux.

**Comment by hubutui** [comment]
I see python2.7 in the binary tarball. Hmm, why not try this https://github.com/pcxod/olex2/issues/1?

**Comment by SamLukeYes** [comment]
> I see python2.7 in the binary tarball. Hmm, why not try this https://github.com/pcxod/olex2/issues/1?

I'm not confident enough to package such a complex program by myself... Anyway, I think I'll have a try later today if no one do that earlier than me :)

**Comment by hubutui** [comment]
You could try to build it from the source, if you manage to do it, then create a `PKGBUILD` file.

**Comment by hubutui** [comment]
try this [Makefile.txt](https://github.com/archlinuxcn/repo/files/5504314/Makefile.txt). I could build olex2 now, but I don't know how to use this program, not sure if it works as expected.

**Comment by SamLukeYes** [comment]
> try this [Makefile.txt](https://github.com/archlinuxcn/repo/files/5504314/Makefile.txt). I could build olex2 now, but I don't know how to use this program, not sure if it works as expected. BTW, `make install` would not work, you need to manually copy the necessary files to destination dir if you wanna create pkg for ArchLinux.

We might need to package `cctbx` first. There's only an ugly package called `cctbx-bin` on AUR.

**Comment by hubutui** [comment]
I'm not sure. olex2's doc is poor, I don't get a build instruction on Linux.

**Comment by SamLukeYes** [comment]
> I'm not sure. olex2's doc is poor, I don't get a build instruction on Linux. I think we could safely ignore this warning.

But it is then killed by SIGSEGV! How can we ignore that?:(

**Comment by SamLukeYes** [comment]
According to https://github.com/pcxod/olex2/issues/1#issuecomment-723440202, I think building Olex2 automatically from a PKGBUILD is too difficult. What about making a package with the binary release?

**Comment by hubutui** [comment]
try [this](https://gist.github.com/hubutui/cc41ab1753568d34522da69ce7e0b407), I assume you know how to create a pkg for ArchLinux, if not, check arch wiki. I don't get SIGSEGV error issue.

**Comment by hubutui** [comment]
If you would like to repack a binary pkg, you need to make sure it works, and check also its deps.

**Comment by SamLukeYes** [comment]
> try [this](https://gist.github.com/hubutui/cc41ab1753568d34522da69ce7e0b407), I assume you know how to create a pkg for ArchLinux, if not, check arch wiki. I don't get SIGSEGV error issue.

Did you upload a wrong Makefile? It doesn't match the sha512sum in the PKGBUILD.

**Comment by hubutui** [comment]
why don't you update the checksum?

**Comment by SamLukeYes** [comment]
> why don't you update the checksum?

I tried skipping checksum, but still got the SIGSEGV issue.

**Comment by hubutui** [comment]
https://github.com/pcxod/olex2/issues/1#issuecomment-722857000, Oh, you're using Manjaro. I know nothing about it. Sorry, there is nothing I could do.

**Comment by SamLukeYes** [comment]
> [pcxod/olex2#1 (comment)](https://github.com/pcxod/olex2/issues/1#issuecomment-722857000), Oh, you're using Manjaro. I know nothing about it. Sorry, there is nothing I could do.

Well, one more thing to confirm: the two Makefiles are actually the same, but only Manjaro has this issue, right?

**Comment by hubutui** [comment]
you could always run diff to find the difference between this `Makefile` and the one from olex2 source.

**Comment by SamLukeYes** [comment]
> you could always run diff to find the difference between this `Makefile` and the one from olex2 source.

Yes, I checked the difference with `diff`.

Let me figure out the point. You have posted your Makefile twice. When you posted it for the second time with the PKGBUILD, I thought you had modified the Makefile again to solve the SIGSEGV issue, but later found it the same as the former one. Since its sha512sum doesn't match, I doubted if you had sent a wrong Makefile. Actually you didn't got the SIGSEGV issue, but only meant to send the PKGBUILD to help me reproduce your making process, right? As far as I know now, I might have to rebuild `wxgtk` on Manjaro, as mentioned in https://github.com/pcxod/olex2/issues/1#issuecomment-723440202.
